android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java

```diff
@@ -176,6 +150,41 @@ public void onResponseReceived(long durationMillis) {
       });
     }
 
+    private boolean checkNeedsShortening() {
```

Comment:
I'd accept the encodedUrl as a parameter here (that way you won't have to encode it twice).